File: scala/1702.A.scala
import scala.io.StdIn.readLine
import scala.math.BigInt

object Main extends App {
  val t = readLine().trim.toInt
  for (_ <- 1 to t) {
    val s = readLine().trim
    val n = BigInt(s)
    val power = BigInt(10).pow(s.length - 1)
    println(n - power)
  }
}

File: perl/1702.A.pl
#!/usr/bin/perl
use strict;
use warnings;
use bigint;   # enable arbitrary-precision integers

my $t = <STDIN>;
chomp $t;

for (1 .. $t) {
    my $n_str = <STDIN>;
    chomp $n_str;

    my $len   = length $n_str;
    my $power = 10 ** ($len - 1);   # bigint handles huge exponents
    my $n     = $n_str;              # automatically converted to bigint

    print $n - $power, "\n";
}
